File: core/CompareVersionNumbers/CompareVersionNumbers.py
def compareVersion(version1: str, version2: str) -> int:
    v1 = [int(item) for item in version1.split(".")]
    v2 = [int(item) for item in version2.split(".")]
    len1, len2 = len(v1), len(v2)
    for index in range(max(len1, len2)):
        val1 = v1[index] if index < len1 else 0
        val2 = v2[index] if index < len2 else 0
        if val1 != val2:
            return 1 if val1 > val2 else -1

    return 0

File: core/CompareVersionNumbers/CompareVersionNumbers.java
package com.dbc;

public class CompareVersionNumbers {
    public int compareVersion(String version1, String version2) {
        String[] v1 = version1.split("\\.");
        String[] v2 = version2.split("\\.");
        for (int i = 0; i < Math.max(v1.length, v2.length); i++) {
            int val1 = i < v1.length ? Integer.parseInt(v1[i]) : 0;
            int val2 = i < v2.length ? Integer.parseInt(v2[i]) : 0;
            if (val1 != val2) return val1 < val2 ? -1 : 1;
        }
        return 0;
    }
}
